backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/UpdateClusterUpgradeCommand.java

```diff
@@ -0,0 +1,79 @@
+package org.ovirt.engine.core.bll;
+
+import javax.inject.Inject;
+
+import org.ovirt.engine.core.bll.context.CommandContext;
+import org.ovirt.engine.core.common.AuditLogType;
+import org.ovirt.engine.core.common.action.ClusterUpgradeParameters;
+import org.ovirt.engine.core.common.businessentities.Cluster;
+import org.ovirt.engine.core.common.errors.EngineMessage;
+import org.ovirt.engine.core.compat.Guid;
+import org.ovirt.engine.core.dao.ClusterDao;
+import org.ovirt.engine.core.utils.transaction.TransactionSupport;
+
+@NonTransactiveCommandAttribute(forceCompensation = true)
+public class UpdateClusterUpgradeCommand<T extends ClusterUpgradeParameters> extends ClusterCommandBase<T> {
+
+    @Inject
+    private ClusterDao clusterDao;
+
+    private Cluster cluster;
+
+    @Override
+    protected void init() {
+        cluster = clusterDao.get(getCluster().getId());
+    }
+
+    public UpdateClusterUpgradeCommand(T parameters, CommandContext commandContext) {
+        super(parameters, commandContext);
+    }
+
+    protected UpdateClusterUpgradeCommand(Guid commandId) {
+        super(commandId);
+    }
+
+    @Override
+    protected void executeCommand() {
+        TransactionSupport.executeInNewTransaction(() -> {
+            int progress = getParameters().getUpgradeProgress();
+            boolean updated = clusterDao.setUpgradeProgress(cluster.getId(), progress);
+
+            if (updated) {
+                log.info("Set upgrade progress to {} on cluster '{}'.", progress, cluster.getId());
+            } else {
+                String msg = String.format("Failed to set upgrade progress on cluster '%s'.", cluster.getId());
+                log.error(msg);
+                getReturnValue().getExecuteFailedMessages().add(msg);
+            }
+            setActionReturnValue(updated);
+            setSucceeded(updated);
+            return null;
+        });
+    }
+
+    @Override
+    public AuditLogType getAuditLogTypeValue() {
+        return getSucceeded()
+                ? AuditLogType.USER_UPDATE_CLUSTER_UPGRADE
+                : AuditLogType.USER_UPDATE_CLUSTER_UPGRADE_FAILED;
+    }
+
+    @Override
+    protected boolean validate() {
+        boolean upgradeStarted = cluster.isUpgradeRunning();
+        if (!upgradeStarted) {
+            return failValidation(EngineMessage.ACTION_TYPE_FAILED_CLUSTER_UPGRADE_NOT_STARTED);
+        }
+
+        // TODO: Validate value of progress
```

Comment:
one check that could make our life easier:) `next progress >= previous progress`

Reply:
I never assume that my next step of progress will move my percent complete ahead!  I'm ok excluding the "next progress must be larger than current progress" rule.